### ISSUE TITLE
PGPKeyring should be able to write GPG compatible keyring files to disk

### ISSUE THREAD

**Issue by Commod0re** [SecurityInnovation/PGPy#61]
Once keys are loaded and GPG Trust packets have been added to them, PGPKeyring objects should be able to write themselves to disk.

This will require two (maybe three) attributes:
- A path to write the public keyring (containing only public keys)
- A path to write the secret keyring (containing only private keys)
- It may be necessary (or useful) also to write a trustdb file for GPG to not complain. RFC 4880 does not specify anything about the trustdb; this is specific to GnuPG's implementation.

For unit testing, GPG should be used to load the saved keyring files, and verify the contents.

**Comment by anarcat** [comment]
see also #148

**Comment by anarcat** [comment]
couldn't this be just implemented by piping through `gpg --import`? i am not sure it is wise to start messing around with the gpg storage, in fact it's discourage by GPG developers...

**Comment by Commod0re** [comment]
A GPG keyring is mostly just one or more keys concatenated together without the ASCII armoring, though. It's not necessarily to interact with a given user's actual keyring that they would use through GPG otherwise, but maybe someone might want to import or export files in that format for other purposes (e.g. the files in /etc/apt/trusted.gpg.d/, for example)

Maybe you're right and it's not worth going too far in emulating that storage format, though

**Comment by anarcat** [comment]
that's the format for GnuPG 1.4, but that changed in GPG 2.x (see #148), so it's not that simple... 

i believe that the binary format should be compatible already, no?

**Comment by Commod0re** [comment]
2.0 still used the older format, and 2.x >= 2.1 can import import from them at least. I think you're right about that, though - I'm not sure if not having the GPG trust packets would break anything in cases where those files have more than one key? But maybe PGPy doesn't have to care about that.

I'll have to think about this for a bit, but I think you're making a very good point here regardless

**Comment by dkg** [comment]
I agree with anarcat here -- please don't try to fiddle around with whatever GnuPG puts in ~/.gnupg (or even assume that things are in ~/.gnupg).  gpg provides an `--import` and `--export` interface, which is what should be used for interacting with gpg's key storage.

**Comment by Commod0re** [comment]
This doesn't have to mean messing with currently active GPG files, though. GPG can specify alternate paths for its keyring files for various purposes, and the only difference between an exported keyset and a keyring file are the trust packets, which are not typically exported.

There's nothing about supporting this that inherently means touching GPG's current configuration files. I'm not sure that the potential for someone using it on active GPG files is good reason to avoid it.
